server/channels/store/sqlstore/channel_store.go

```diff
@@ -1961,18 +1967,10 @@ func (s SqlChannelStore) PatchMultipleMembersNotifyProps(members []*model.Channe
 		builder = builder.Set("NotifyProps", jsonExpr)
 	}
 
-	sqlUpdate, args, err := builder.Where(whereClause).ToSql()
-	if err != nil {
-		return nil, errors.Wrap(err, "PatchMultipleMembersNotifyProps_ToSQL")
-	}
+	builder = builder.Set("LastUpdateAt", model.GetMillis())
 
-	transaction, err := s.GetMasterX().Beginx()
+	result, err := transaction.ExecBuilder(builder.Where(whereClause))
```

Comment:
Propose we don't modify the `builder` at the same time we execute it, but instead apply it immediately to the `builder` up after we first define it (around line 1951). It's not necessary for the builder to follow the same order as the final SQL result.
```go
	builder = builder.Where(whereClause)

	// Update the channel members
        ...

	result, err := transaction.ExecBuilder(builder)
```

Reply:
Agree :+1:

Reply:
I'd rather keep it in the order of the SQL query since that's the order I'd expect the code to be, but I can move that out for clarity. Strictly speaking though, we're not actually modifying the builder there since the builder copies itself on write if that's what you were worried about.

Reply:
> I'd rather keep it in the order of the SQL query since that's the order I'd expect the code to be

I understand this inclination, but this isn't the pattern we're using elsewhere in the store (AFAIK). In practice, this ends up being impossible anyway in complicated scenarios, so it's not a standard worth trying to preserve. I also think it's /more/ confusing to split up the `whereClause` definition and usage.

>  Strictly speaking though, we're not actually modifying the builder there since the builder copies itself on write if that's what you were worried about.

In fact I think it's /worse/ that we didn't modify the builder, again because it's "surprising" relative to the rest of the store code. There are often valid uses cases for reusing a builder, and while this isn't one of them, it would be surprising to try to reuse `builder` and discover only part of the SQL query present.